data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisService.java

```diff
@@ -139,6 +142,8 @@ public Scheduler getScheduler(final Buffer<Record<Event>> buffer) {
         while (scheduler == null && numRetries++ < kinesisSourceConfig.getMaxInitializationAttempts()) {
             try {
                 scheduler = createScheduler(buffer);
+            } catch (SdkClientException | KinesisClientLibDependencyException | ThrottlingException ex) {
+                LOG.error(NOISY, "Caught exception when initializing KCL Scheduler due to {}. Will retry", ex.getMessage());
```

Comment:
This does not always retry as you have a `numRetries` above. Maybe drop the "Will retry" text or say "The number of retries remaining is N". You could count down instead of counting up in the loop above.

Reply:
Thanks @dlvenable . I will drop the `Will retry` part of the statement.